pkg/cmd/roachtest/operations/backup_restore.go

```diff
@@ -69,64 +75,162 @@ outer:
 		return nil
 	}
 
-	backupTS := timeutil.Now().Add(-1 * time.Minute).UTC().Format(time.DateTime)
-
-	o.Status(fmt.Sprintf("backing up table district in db %s", dbName))
+	o.Status(fmt.Sprintf("backing db %s (full)", dbName))
 	bucket := fmt.Sprintf("gs://%s/operation-backup-restore/%d/?AUTH=implicit", testutils.BackupTestingBucket(), timeutil.Now().UnixNano())
 
-	_, err = conn.ExecContext(ctx, fmt.Sprintf("BACKUP TABLE %s.district TO '%s' AS OF SYSTEM TIME '%s'", dbName, bucket, backupTS))
+	backupTS := hlc.Timestamp{WallTime: timeutil.Now().Add(-10 * time.Second).UTC().UnixNano()}
+	_, err = conn.ExecContext(ctx, fmt.Sprintf("BACKUP DATABASE %s INTO '%s' AS OF SYSTEM TIME '%s'", dbName, bucket, backupTS.AsOfSystemTime()))
 	if err != nil {
 		o.Fatal(err)
 	}
 
-	restoreDBName := fmt.Sprintf("backup_restore_op_%d", rng.Int63())
-	_, err = conn.ExecContext(ctx, fmt.Sprintf("CREATE DATABASE %s", restoreDBName))
-	if err != nil {
-		o.Fatal(err)
+	if !online {
+		for i := range 24 {
+			o.Status(fmt.Sprintf("backing up db %s (incremental layer %d)", dbName, i))
+			// Update backupTS to match the latest layer.
+			backupTS = hlc.Timestamp{WallTime: timeutil.Now().Add(-10 * time.Second).UTC().UnixNano()}
+			_, err = conn.ExecContext(ctx, fmt.Sprintf("BACKUP DATABASE %s INTO LATEST IN '%s' AS OF SYSTEM TIME '%s'", dbName, bucket, backupTS.AsOfSystemTime()))
+			if err != nil {
+				o.Fatal(err)
+			}
+		}
 	}
 
-	o.Status(fmt.Sprintf("restoring table district into db %s", restoreDBName))
-	_, err = conn.ExecContext(ctx, fmt.Sprintf("RESTORE TABLE %s.district FROM '%s' AS OF SYSTEM TIME '%s' WITH OPTIONS (into_db = '%s', skip_missing_foreign_keys)", dbName, bucket, backupTS, restoreDBName))
+	restoreDBName := fmt.Sprintf("backup_restore_op_%d", rng.Int63())
 
-	if err != nil {
-		o.Fatal(err)
+	onlineStr := "online"
+	if !online {
+		onlineStr = "offline"
 	}
+	o.Status(fmt.Sprintf("restoring %s into db %s", onlineStr, restoreDBName))
+	startTime := timeutil.Now()
+	if !online {
+		o.Status("beginning offline restore")
+		_, err = conn.ExecContext(ctx, fmt.Sprintf("RESTORE DATABASE %s FROM LATEST IN '%s' AS OF SYSTEM TIME '%s' WITH OPTIONS (new_db_name = '%s')", dbName, bucket, backupTS.AsOfSystemTime(), restoreDBName))
```

Comment:
i don't think you need these AOST clauses in either statement. We'll just restore from the latest backup in the chain in both. in fact, the OR stmt may fail? not sure